src/shared/context/TableContext/TableContext.test.tsx

```diff
@@ -0,0 +1,1241 @@
+import { NoContextProviderError } from '@/lib/errors/ContextError'
+import { cleanup, fireEvent, render, screen, waitFor } from '@testing-library/react'
+import React, { FC, useContext } from 'react'
+import { afterEach, describe, expect, it } from 'vitest'
+import { useTableActionsContext, useTableContext, withTableContext } from '.'
+import { TableActionsContext } from './TableActionsContext'
+import { TableContext, initialState } from './TableContext'
+
+afterEach(() => {
+  cleanup()
+})
+
+// Helper function to parse JSON from test elements
+const parseTestElementData = (testId: string) => {
+  const element = screen.getByTestId(testId)
+  return JSON.parse(element.textContent || '{}')
+}
+
+describe('TableContext Hook Usage', () => {
+  describe('useTableContext', () => {
+    const ContextDisplayComponent: FC = () => {
+      const context = useTableContext()
+      return <div data-testid="context-data">{JSON.stringify(context)}</div>
+    }
+    it('throws error when used outside TableProvider', () => {
+      expect(() => render(<ContextDisplayComponent />)).toThrow(
+        new NoContextProviderError('useTableContext', 'TableProvider'),
+      )
+    })
+
+    it('provides initial state when wrapped with TableProvider', async () => {
+      const WrappedComponent = withTableContext(ContextDisplayComponent)
+      render(<WrappedComponent />)
+
+      await waitFor(() => {
+        expect(screen.getByTestId('context-data')).toBeInTheDocument()
+      })
+
+      const contextData = parseTestElementData('context-data')
+      expect(contextData).toEqual(initialState)
+    })
+  })
+
+  describe('useTableActionsContext', () => {
+    const DispatcherTestComponent: FC = () => {
+      useTableActionsContext()
+      return <div data-testid="dispatcher-available">Dispatcher is available</div>
+    }
+
+    it('throws error when used outside TableProvider', () => {
+      expect(() => render(<DispatcherTestComponent />)).toThrow()
+    })
+
+    it('provides dispatch function when wrapped with TableProvider', async () => {
+      const WrappedComponent = withTableContext(DispatcherTestComponent)
+      render(<WrappedComponent />)
+
+      await waitFor(() => {
+        expect(screen.getByTestId('dispatcher-available')).toBeInTheDocument()
+      })
+    })
+  })
+})
+
+describe('TableProvider Integration', () => {
+  const DirectContextAccessComponent: FC = () => {
+    const context = useContext(TableContext)
+    const dispatcher = useContext(TableActionsContext)
+
+    return (
+      <div>
+        <div data-testid="direct-context">{context ? JSON.stringify(context) : 'null'}</div>
+        <div data-testid="direct-dispatcher">{typeof dispatcher === 'function' ? 'available' : 'null'}</div>
+      </div>
+    )
+  }
+  it('provides both context and dispatcher together', async () => {
```

Comment:
how do you validate that the context was provided? You are looking for the id that was included manually, I dont see that any expect, validates this information